Essentials/src/com/earth2me/essentials/commands/Commandkickall.java

```diff
@@ -20,6 +20,9 @@ public void run(final Server server, final CommandSource sender, final String co
 
         for (Player onlinePlayer : ess.getOnlinePlayers()) {
             if (!sender.isPlayer() || !onlinePlayer.getName().equalsIgnoreCase(sender.getPlayer().getName())) {
+                if (onlinePlayer.isAuthorized("essentials.kickall.exempt")) {
+                    throw new Exception();
```

Comment:
This shouldn't throw an exception - that will completely interrupt the kickall.

Instead, `onlinePlayer.kickPlayer(kickReason)` should only be called if the player does *not* have the `essentials.kickall.exempt` permission.